### PR TITLE
Qa to Master: dev mode 

### PR DESCRIPTION
**Pull request by evelinec**
**Required changes:**
- use the ideal(simplified) pom.xml
- update the Readme content to use dev mode
  - keep the flow same 
- convert `*Test` class to `*IT` (N/A)
- convert to use JUnit 5 (N/A)
  - use JUnit5 test order if need
- make sure to correctly configure the ports in test classes, server.xml, and pom.xml
- update travis test
   - `mvn install` cannot be used to verify anymore

- Also fixed issue #42

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -287,7 +288,7 @@ include::finish/src/main/frontend/src/app/app.component.html[]
 
 The template contains a [hotspot=artistsDiv]`div` element that is enumerated using the
```
Comment (Charlotte-Holt) [inline README.adoc]: Update
"The template contains a [hotspot=artistsDiv]`div` element that is enumerated using the `*ngFor` directive."
To
"The template contains a [hotspot=artistsDiv`div` element that is enumerated by using the `*ngFor` directive."

File: README.adoc [hunk 2]
```diff
@@ -313,6 +314,11 @@ bar wrote 1 albums:
 dj wrote 0 albums:
 ----
 
+If you use the `curl` command to visit the web application root URL,
+you will only see the application root page in HTML. The Angular framework
```
Comment (Charlotte-Holt) [inline README.adoc]: Update
"If you use the `curl` command to visit the web application root URL, you will only see the application root page in HTML."
To
"If you use the `curl` command to go to the web application root URL, you see only the application root page in HTML."

File: README.adoc (hunk 1 above)
Comment (evelinec) [reply README.adoc]: Updated.

File: README.adoc (hunk 2 above)
Comment (evelinec) [reply README.adoc]: Updated

**Comment by DannyS03** [comment]
- tested/verified on `master`.
- LGTM.

**Pull request merged**